components/d2l-applied-filters/d2l-applied-filters.js

```diff
@@ -0,0 +1,237 @@
+import { LitElement, css, html } from 'lit-element';
+import { LocalizeStaticMixin } from '@brightspace-ui/core/mixins/localize-static-mixin.js';
+
+import '@brightspace-ui-labs/multi-select/multi-select-list';
+import '@brightspace-ui-labs/multi-select/multi-select-list-item';
+import { bodyCompactStyles } from '@brightspace-ui/core/components/typography/styles.js';
+import { getComposedChildren } from '@brightspace-ui/core/helpers/dom';
+import { RtlMixin } from '@brightspace-ui/core/mixins/rtl-mixin.js';
+
+const DROPDOWN_NAME = 'D2L-FILTER-DROPDOWN';
+const DROPDOWN_CATEGORY_NAME = 'D2L-FILTER-DROPDOWN-CATEGORY';
+const DROPDOWN_OPTION_NAME = 'D2L-FILTER-DROPDOWN-OPTION';
+
+class D2lAppliedFilters extends RtlMixin(LocalizeStaticMixin(LitElement)) {
+	static get properties() {
+		return {
+			for: { type: String },
+			_entries: { type: Array },
+			_selectedEntries: { type: Array },
+			_target: { type: Object },
+		};
+	}
+
+	static get styles() {
+		return [bodyCompactStyles, css`
+			:host {
+				display: block;
+			}
+
+			d2l-labs-multi-select-list {
+				flex: 1;
+			}
+
+			.d2l-applied-filters-wrapper {
+				display: flex;
+			}
+
+			.d2l-applied-filters-applied-filters-label {
+				margin-right: 0.25rem;
+				display: inline-block;
+				padding-top: 0.3rem;
+				font-weight: bold;
+			}
+
+			:host([dir="rtl"]) .d2l-applied-filters-applied-filters-label {
+				margin-right: 0;
+				margin-left: 0.25rem;
+			}
+
+			.d2l-applied-filters-no-applied-filters-label {
+				display: inline-block;
+				padding-top: 0.3rem;
+				font-style: italic;
+				color: var(--d2l-color-corundum);
+			}
+
+			:host([hidden]) {
+				display: none;
+			}
+		`];
+	}
+
+	static get resources() {
+		return {
+			'ar': {
+			},
+			'de': {
+			},
+			'en': {
+				appliedFilters: 'Applied Filters:',
+				noActiveFilters: 'No active filters'
+			},
+			'es': {
+			},
+			'fi': {
+			},
+			'fr': {
+			},
+			'ja': {
+			},
+			'ko': {
+			},
+			'nb': {
+			},
+			'nl': {
+			},
+			'pt': {
+			},
+			'sv': {
+			},
+			'tr': {
+			},
+			'zh': {
+			},
+			'zh-tw': {
+			}
+		};
+	}
+
+	constructor() {
+		super();
+		this._clearSelected = this._clearSelected.bind(this);
+		this._setSelectedOptions = this._setSelectedOptions.bind(this);
+		this._update = this._update.bind(this);
+	}
+
+	firstUpdated() {
+		this._update();
+	}
+
+	updated(changedProperties) {
+		if (Object.keys(changedProperties).indexOf('for') > -1) {
+			this._update();
+		}
+	}
+
+	_update(e) {
+		this._setOptions(e && e.detail.categoryKey);
+		this._setSelectedOptions();
+	}
+
+	_filterOptionChanged() {
+		this._setSelectedOptions();
+	}
+
+	_multiSelectItemDeleted(e) {
+		const entryIndex = Number.parseInt(e.path[0].attributes.index.value);
+		const entry = this._selectedEntries[entryIndex];
+		entry.dispatchEvent(new CustomEvent('d2l-menu-item-select'));
```

Comment:
What would the consumer use this event for?

Reply:
The deletion was for the multi-select list, I put up another commit to not do this with an event since it was unintuitive